src/test/java/tech/stackable/trino/opa/OpaAuthorizerTest.java

```diff
@@ -0,0 +1,166 @@
+                + attempts + " attempts made at " + timeoutMs + "ms each)");
+    }
+
+    @BeforeAll
+    public static void setupTrino() throws IOException, InterruptedException {
+        InetSocketAddress opaSocket = findAvailableTcpPort();
+        opaServer = new ProcessBuilder("opa", "run", "--server", "--addr",
+                opaSocket.getHostString() + ":" + opaSocket.getPort()).inheritIO().start();
+        awaitSocketOpen(opaSocket, 100, 200);
+        opaServerUri =
+                URI.create("http://" + opaSocket.getHostString() + ":" + opaSocket.getPort() + "/");
+
+        QueryIdGenerator idGen = new QueryIdGenerator();
+        Identity identity = Identity.forUser("test-user").build();
+        SessionPropertyManager sessionPropertyManager = new SessionPropertyManager();
+        Session session = Session.builder(sessionPropertyManager)
+                .setQueryId(idGen.createNextQueryId()).setIdentity(identity).build();
+        trinoServer = TestingTrinoServer.builder()
+                .setSystemAccessControls(Collections.singletonList(new OpaAuthorizer(opaServerUri.resolve("v1/data/trino/"))))
+                .build();
+        trinoClient = new TestingTrinoClient(trinoServer, session);
+    }
+
+    @AfterAll
+    public static void teardownTrino() throws IOException, InterruptedException {
+        try {
+            if (opaServer != null) {
+                opaServer.destroy();
+            }
+        } finally {
+            try {
+                if (trinoClient != null) {
+                    trinoClient.close();
+                }
+            } finally {
+                if (trinoServer != null) {
+                    trinoServer.close();
+                }
+            }
+        }
+    }
+
+    private String stringOfLines(String... lines) {
+        StringBuilder out = new StringBuilder();
+        for (String line : lines) {
+            out.append(line);
+            out.append("\r\n");
+        }
+        return out.toString();
+    }
+
+    private void submitPolicy(String... policyLines) throws IOException, InterruptedException {
+        HttpClient httpClient = HttpClient.newHttpClient();
+        HttpResponse<String> policyRes =
+                httpClient.send(
+                        HttpRequest.newBuilder(opaServerUri.resolve("v1/policies/trino"))
+                                .PUT(HttpRequest.BodyPublishers
+                                        .ofString(stringOfLines(policyLines)))
+                                .header("Content-Type", "text/plain").build(),
+                        HttpResponse.BodyHandlers.ofString());
+        assertEquals(policyRes.statusCode(), 200, "Failed to submit policy: " + policyRes.body());
+    }
+
+    @Test
+    public void testShouldAllowQueryIfDirected() throws IOException, InterruptedException {
+        submitPolicy("package trino", "default can_execute_query = true",
```

Comment:
I have no strong objection to doing things this way, but it seems a bit cumbersome to create an indented multiline string.
Would maybe using test resources and reading from a file work as well?
That would have the drawback of not having the rule readable in the test though, which is significant ...

Reply:
Yeah.. it's definitely not pretty. But pulling in something like Scala or requiring a newer JDK just for multiline strings also feels overkill.. :/

Reply:
fine for me